src/ibtracert.c

```diff
@@ -825,57 +867,83 @@ int main(int argc, char **argv)
 			       ibd_dest_type, ibd_sm_id, srcport) < 0)
-		IBEXIT("can't resolve source port %s", argv[0]);
+			IBEXIT("can't resolve source port %s", argv[0]);
 
-	if (resolve_portid_str(ibd_ca, ibd_ca_port, &dest_portid, argv[1],
+		if (resolve_portid_str(ibd_ca, ibd_ca_port, &dest_portid, dstid,
 			       ibd_dest_type, ibd_sm_id, srcport) < 0)
-		IBEXIT("can't resolve destination port %s", argv[1]);
-
-	if (ibd_dest_type == IB_DEST_DRPATH) {
-		if (resolve_lid(&src_portid, NULL) < 0)
-			IBEXIT("cannot resolve lid for port \'%s\'",
-				portid2str(&src_portid));
-		if (resolve_lid(&dest_portid, NULL) < 0)
-			IBEXIT("cannot resolve lid for port \'%s\'",
-				portid2str(&dest_portid));
-	}
+			IBEXIT("can't resolve destination port %s", argv[1]);
+
+		if (ibd_dest_type == IB_DEST_DRPATH) {
+			if (resolve_lid(&src_portid, NULL) < 0)
+				IBEXIT("cannot resolve lid for port \'%s\'",
+					portid2str(&src_portid));
+			if (resolve_lid(&dest_portid, NULL) < 0)
+				IBEXIT("cannot resolve lid for port \'%s\'",
+					portid2str(&dest_portid));
+		}
 
-	if (dest_portid.lid == 0 || src_portid.lid == 0) {
-		IBWARN("bad src/dest lid");
-		ibdiag_show_usage();
-	}
+		if (dest_portid.lid == 0 || src_portid.lid == 0) {
+			IBWARN("bad src/dest lid");
+			ibdiag_show_usage();
+		}
 
-	if (ibd_dest_type != IB_DEST_DRPATH) {
-		/* first find a direct path to the src port */
-		if (find_route(&my_portid, &src_portid, 0) < 0)
-			IBEXIT("can't find a route to the src port");
+		if (ibd_dest_type != IB_DEST_DRPATH) {
+			/* first find a direct path to the src port */
+			if (find_route(&my_portid, &src_portid, 0) < 0)
+				IBEXIT("can't find a route to the src port");
 
-		src_portid = my_portid;
-	}
+			src_portid = my_portid;
+		}
 
-	if (!multicast) {
-		if (find_route(&src_portid, &dest_portid, dumplevel) < 0)
-			IBEXIT("can't find a route from src to dest");
-		exit(0);
-	} else {
-		if (mlid < 0xc000)
-			IBWARN("invalid MLID; must be 0xc000 or larger");
-	}
+		if (!multicast) {
+			if (find_route(&src_portid, &dest_portid, dumplevel) < 0)
+				IBEXIT("can't find a route from src to dest");
+		} else {
+			if (mlid < 0xc000)
+				IBWARN("invalid MLID; must be 0xc000 or larger");
 
-	if (!(target_portguid = find_target_portguid(&dest_portid)))
-		IBEXIT("can't reach target lid %d", dest_portid.lid);
+			if (!(target_portguid = find_target_portguid(&dest_portid)))
+				IBEXIT("can't reach target lid %d", dest_portid.lid);
 
-	if (!(endnode = find_mcpath(&src_portid, mlid)))
-		IBEXIT("can't find a multicast route from src to dest");
+			if (!(endnode = find_mcpath(&src_portid, mlid)))
+				IBEXIT("can't find a multicast route from src to dest");
 
-	/* dump multicast path */
-	dump_mcpath(endnode, dumplevel);
+			/* dump multicast path */
+			dump_mcpath(endnode, dumplevel);
+		}
 
+		if (ports_fd) {
+			my_portid.drpath.cnt = 0;
+			src_portid.drpath.cnt = 0;
+     	   		if (fgets(ports_buffer, sizeof(ports_buffer), ports_fd)) {
```

Comment:
Extra tab before if ?

Reply:
Are you saying I should add a tab, or you think there is an extra tab ?
I don't see it.

Reply:
It looks like there is extra tab from the above as the if is not lined up with the previous line.